### PR TITLE
Testable phpminimumversion

### PR DESCRIPTION
**Pull request by mistermuckle**
Moved phpminimumversionlib into a class and added tests.

I retained both global functions `moodle_require_minumum_php_version()` and `moodle_minimum_php_version_is_met()`, even though the former has been replaced by calls to he new class, and the latter isn’t used anywhere. I’m not sure whether these global functions get called by Moodle plugins, so I wanted to err toward safety.

### CODE REVIEW

File: Tests/lib/PhpMinimumVersionTest.php [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+use Moodle\Egress\EgressException;
+
+class PhpMinimumVersionTest extends \PHPUnit\Framework\TestCase
+{
+    const ERROR_MESSAGE = 'Moodle 3.4 or later requires at least PHP 7.0.0 (currently using version 5.6.8).' . PHP_EOL . 'Some servers may have multiple PHP versions installed, are you using the correct executable?' . PHP_EOL;
+
+    public function testRequireMinimumPhpVersionDoesNothingIfPhpVersionIs7()
+    {
+        $phpMinimumVersion = new PhpMinimumVersion('7.0.0', false, STDERR);
```
Comment (CarwynNelson) [inline Tests/lib/PhpMinimumVersionTest.php]: could we extract these usages of `false` out into a constant or variable in the test? It's not immediately obvious to the reader of the test what this value actually **is**

File: Tests/lib/PhpMinimumVersionTest.php [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    public function testRequireMinimumPhpVersionOutputsMessageIfPhpVersionIsLessThanSeven()
+    {
+        $phpMinimumVersion = new PhpMinimumVersion('5.6.8', false, STDERR);
+        $this->expectOutputString(self::ERROR_MESSAGE);
+
+        try {
+            $phpMinimumVersion->requireMinimumPhpVersion();
+        } catch (EgressException $e) {
```
Comment (CarwynNelson) [inline Tests/lib/PhpMinimumVersionTest.php]: This try-catch seems a little odd to me. The test is saying that it expects output to be produced. If the exception is thrown surely it means that the output has not been created and so the test should fail?

File: Tests/lib/PhpMinimumVersionTest.php [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    }
+
+    public function testMinimumPhpVersionIsMetOutputsMessageIfPhpVersionIsLessThanSevenAndHaltexecutionIsTrue()
+    {
+        $phpMinimumVersion = new PhpMinimumVersion('5.6.8', false, STDERR);
+        $this->expectOutputString(self::ERROR_MESSAGE);
+
+        try {
+            $phpMinimumVersion->minimumPhpVersionIsMet(true);
+        } catch (EgressException $e) {
```
Comment (CarwynNelson) [inline Tests/lib/PhpMinimumVersionTest.php]: can the "and halts execution" bit be handled by the built in phpunit methods for exceptions? I believe you can either call `$this->setExpectedException()` or the `@expectedException` annotation.

**Comment by CarwynNelson** [comment]
Really liking the pulling of this component out into a class :) Very clean.

**Comment by mistermuckle** [comment]
I’ve cleaned up the tests, and added constants to make everything clearer.

**Comment by CarwynNelson** [comment]
I wouldn't worry too much about the failing build for now. It is passing in my branch but it doesn't appear to be doing much.

I've ran the tests locally and they look good to me.

I'm going to look at the phpunit config to get it to run these tests (and any new ones that you add).

**Comment by mistermuckle** [comment]
Ok then I’ll go ahead and merge. Thanks.

**Pull request merged**